### PR TITLE
fixes/pt-5/authentication

### PR DESCRIPTION
**Pull request by dungeon-slave**
https://github.com/dungeon-slave/food_app/assets/70040363/867946fa-ac14-4ee1-b432-f58dc82e7b3a

migrate to the new auto_route
make responsive app bars
create text_field core widget
end sign-in/sign-out layout
fix navigation
typo fixes

### CODE REVIEW

File: core_ui/lib/src/widgets/app_button.dart [hunk 1]
```diff
@@ -15,7 +15,8 @@ class AppButton extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     return ElevatedButton(
-      style: ElevatedButton.styleFrom(backgroundColor: Theme.of(context).indicatorColor),
+      style: ElevatedButton.styleFrom(
+          backgroundColor: Theme.of(context).indicatorColor),
```
Comment (ibondarovich) [inline core_ui/lib/src/widgets/app_button.dart]: ,

File: domain/lib/usecase/authentication/save_user_usecase.dart [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import 'package:domain/repositories/authentication_repository.dart';
+import 'package:domain/usecase/usecase.dart';
+
+class SaveUserUseCase implements AsyncUseCase<String, void> {
+  final AuthenticationRepository _authenticationRepository;
+
+  const SaveUserUseCase(
+      {required AuthenticationRepository authenticationRepository})
```
Comment (ibondarovich) [inline domain/lib/usecase/authentication/save_user_usecase.dart]: formatting

File: domain/lib/usecase/text_scale/set_text_scale_usecase.dart [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+class SetTextScaleUseCase implements AsyncUseCase<double, void> {
+  final TextScaleRepository _textScaleRepository;
+
+  const SetTextScaleUseCase({
+    required TextScaleRepository textScaleRepository,
+  }) : _textScaleRepository = textScaleRepository;
+
+  @override
+  Future<void> execute(double input) async => await _textScaleRepository.saveTextScale(input);
```
Comment (ibondarovich) [inline domain/lib/usecase/text_scale/set_text_scale_usecase.dart]: formatting

File: features/detailed_page/pubspec.yaml [hunk 4]
```diff
@@ -14,6 +14,10 @@ dependencies:
     path: ../../domain
   core_ui:
     path: ../../core_ui
+  core:
+    path: ../../core
+  auto_route_generator: ^7.3.0
```
Comment (ibondarovich) [inline features/detailed_page/pubspec.yaml]: to dev dependencies

File: features/main_page_screen/pubspec.yaml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  core: 
+    path: ../../core
+  core_ui:
+    path: ../../core_ui
+  domain: 
+    path: ../../domain
+  navigation:
+    path: ../../navigation
+  build_runner: ^2.4.6
+  auto_route_generator: ^7.3.0
```
Comment (ibondarovich) [inline features/main_page_screen/pubspec.yaml]: to dev dependencies

File: features/sign_up_screen/lib/src/bloc/sign_up_bloc.dart [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+        _saveUserUseCase = saveUserUseCase,
+        _checkUserUseCase = checkUserUseCase,
+        _appRouter = appRouter,
+        _authService = authService,
+        super(const SignUpState()) {
+    on<SignUpEvent>(_emailSignUp);
+  }
+
+  Future<void> _emailSignUp(
+      SignUpEvent event, Emitter<SignUpState> emit) async {
```
Comment (ibondarovich) [inline features/sign_up_screen/lib/src/bloc/sign_up_bloc.dart]: ,

File: features/sign_up_screen/lib/src/bloc/sign_up_bloc.dart [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    String userId = await _emailSignUpUseCase.execute(event.data);
+    await _saveUserUseCase.execute(userId);
+    _authService.authenticated = _checkUserUseCase.execute(const NoParams());
+    _appRouter.replace(const HomeRoute());
+    //TODO implement state change
```
Comment (ibondarovich) [inline features/sign_up_screen/lib/src/bloc/sign_up_bloc.dart]: ?

File: navigation/lib/src/app_router/auth_guard.dart [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+    required AuthService authService,
+  }) : _authService = authService;
+
+  @override
+  void onNavigation(NavigationResolver resolver, StackRouter router) {
+    if (_authService.authenticated) {
+      resolver.next();
+    } else {
+      router.pushAndPopUntil(const EmptySign(),
+          predicate: (Route<dynamic> predicate) => false);
```
Comment (ibondarovich) [inline navigation/lib/src/app_router/auth_guard.dart]: ,

**Review by ibondarovich — CHANGES_REQUESTED** [review]

File: features/sign_up_screen/lib/src/bloc/sign_up_bloc.dart (hunk 7 above)
Comment (dungeon-slave) [reply features/sign_up_screen/lib/src/bloc/sign_up_bloc.dart]: fix in next PR

**Pull request merged**